OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -830,6 +839,23 @@ static void ShowStatusBarsDropdown(DropDownButtonWidget dropdown, GameSettings s
 			dropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 500, options.Keys, setupItem);
 		}
 
+		static void ShowDisplaySelectionDropdown(DropDownButtonWidget dropdown, GraphicSettings s)
+		{
+			var options = Game.Renderer.Displays;
+
+			Func<string, ScrollItemWidget, ScrollItemWidget> setupItem = (o, itemTemplate) =>
+			{
+				var item = ScrollItemWidget.Setup(itemTemplate,
+					() => s.VideoDisplay == options[o],
+					() => s.VideoDisplay = options[o]);
+
+				item.Get<LabelWidget>("LABEL").GetText = () => o;
+				return item;
+			};
+
+			dropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 500, options.Keys, setupItem);
```

Comment:
The `options.Keys` here can change to `Enumerable.Range(0, options.Length)` to compensate for changing the type of `Game.Renderer.Displays`.